### DISCUSSION TITLE
Expiry timestamp / rollover indication in the UI

### DISCUSSION THREAD

**Discussion by klochowicz** [get10101/itchysats#1272] [Ideas]
Currently from the perspective of the taker's UI we do not know whether  rollovers worked or not; nor when the user should be online to monitor whether force-closing worked etc.

Eventually we'd like to have an "inbox", but for the time being a field with expiry timestamp should be sufficient.

Thoughts?

**Comment by klochowicz** [comment]
side note - I added an "UI-bug" label, does anyone mind if I just change its name to "UI" ? (2 labels for UIs would be an overkill IMHO ;) )

**Reply by thomaseizinger** [reply]
Go for it.

**Comment by thomaseizinger** [comment]
> Eventually we'd like to have an "inbox", but for the time being a field with expiry timestamp should be sufficient.

I think the discussion roots a bit deeper than that.

1. By nature, CFDs are perpetual, meaning that users should come with the expectation that their positions are open until they close them.
2. Being a L2 product, I think we can mandate from our users to be online 24/7. It is a bit unfortunate but designing secure L2 protocols that can operate while you are temporarily offline is really hard. The Lightning people have been working on watchtowers for years and AFAIK, there is still not solution that doesn't have annoying trade-offs.
3. Assuming (1) and (2), I think it is worthwhile to make a distinction between the funding rounds, i.e. when users get charged fees versus the underlying rollover protocol. There are ways for us to pre-compute the next funding round to for example distribute the load of signing and verifying transactions across a big timeframe without those transactions being active yet (absolute timelocks on `commit` come to my mind).

Once you start thinking about (3), the question of what to display to the user becomes quite hard to answer. Assuming we have the same funding period for everyone, is it good enough to say that the next funding period is in X minutes (which we already display?). One could imagine that we have precomputed the commit transaction already with a timelock so it could in theory be broadcasted literally on the second that the funding period starts.
